gems/pending/util/mount/miq_generic_mount_session.rb

```diff
@@ -457,14 +452,14 @@ def self.raw_disconnect(mnt_point)
     end
   end
 
-  def self.base_mount_point
-    case Sys::Platform::IMPL
-    when :macosx
-      "/Volumes"
-    when :linux
-      "/mnt"
-    else
-      raise "platform not supported"
-    end
+  private
+
+  def settings_mount_point
+    return if @settings[:mount_point].blank? # Check if settings contains the mount_point to use
+    mount_point = File.join(@settings[:mount_point], "miq_#{MiqUUID.new_guid}")
```

Comment:
Since you're already in here @brandondunne, how about remove more logic?

Let ruby create the Temp file for us instead of using MiqUUID, note, you need to run mkdir_p to create the parent directory structure:

```ruby
irb(main):013:0> `ls /Users/joerafaniello/test/temp`
ls: /Users/joerafaniello/test/temp: No such file or directory
=> ""
irb(main):014:0> `ls /Users/joerafaniello/test`
ls: /Users/joerafaniello/test: No such file or directory
=> ""
irb(main):015:0> t = Tempfile.new("miq_", "/Users/joerafaniello/test/temp")
Errno::ENOENT: No such file or directory @ rb_sysopen - /Users/joerafaniello/test/temp/miq_20150901-13559-1el0jhp
	from /Users/joerafaniello/.rubies/ruby-2.2.3/lib/ruby/2.2.0/tempfile.rb:136:in `initialize'
	from /Users/joerafaniello/.rubies/ruby-2.2.3/lib/ruby/2.2.0/tempfile.rb:136:in `open'
	from /Users/joerafaniello/.rubies/ruby-2.2.3/lib/ruby/2.2.0/tempfile.rb:136:in `block in initialize'
	from /Users/joerafaniello/.rubies/ruby-2.2.3/lib/ruby/2.2.0/tmpdir.rb:129:in `create'
	from /Users/joerafaniello/.rubies/ruby-2.2.3/lib/ruby/2.2.0/tempfile.rb:133:in `initialize'
	from (irb):15:in `new'
	from (irb):15
	from /Users/joerafaniello/.rubies/ruby-2.2.3/bin/irb:11:in `<main>'
irb(main):016:0> require 'fileutils'
=> false
irb(main):017:0> FileUtils.mkdir_p("/Users/joerafaniello/test/temp")
=> ["/Users/joerafaniello/test/temp"]
irb(main):018:0> t = Tempfile.new("miq_", "/Users/joerafaniello/test/temp")
=> #<Tempfile:/Users/joerafaniello/test/temp/miq_20150901-13559-mikx2i>
irb(main):019:0> `ls /Users/joerafaniello/test/temp`
=> "miq_20150901-13559-mikx2i\n"
```

I don't remember if other code is expecting a format to the mount pount file name but it's easy to use the miq_ prefix as shown above.

What do you think?